notebooks/ch17/classification_multiclass.md

```diff
@@ -0,0 +1,312 @@
+```
+
+
+![png](classification_multiclass_files/classification_multiclass_7_0.png)
+
+
+Similarly, we build binary classifiers for the remaining classes:
+
+
+```python
+plot_binary(shapes, 'square')
+```
+
+
+![png](classification_multiclass_files/classification_multiclass_9_0.png)
+
+
+
+```python
+plot_binary(shapes, 'circle')
+```
+
+
+![png](classification_multiclass_files/classification_multiclass_10_0.png)
+
+
+We know that the output of the sigmoid function in logistic regression is a probability value from 0 to 1. Then to solve our multiclass classification task, we find the probability of the positive class in each binary classifier and select the class that outputs the highest positive class probability. For example, if we have a new observation with the following values:
+
+$x_1$ | $x_2$
+---   | ---
+3.2   | 2.5
+
+Then our multiclass classifier would fit three binary classifiers to each of the labels of `triangle`, `square`, and `circle`. We extract the positive class probability of each of the three classifiers:
+
+
+```python
+# HIDDEN
+lr_triangle = LogisticRegression(random_state=42)
+lr_triangle.fit(shapes[['$x_1$', '$x_2$']], shapes['$y$'] == 'triangle')
+proba_triangle = lr_triangle.predict_proba([[3.2, 2.5]])[0][1]
+
+lr_square = LogisticRegression(random_state=42)
+lr_square.fit(shapes[['$x_1$', '$x_2$']], shapes['$y$'] == 'square')
+proba_square = lr_square.predict_proba([[3.2, 2.5]])[0][1]
+
+lr_circle = LogisticRegression(random_state=42)
+lr_circle.fit(shapes[['$x_1$', '$x_2$']], shapes['$y$'] == 'circle')
+proba_circle = lr_circle.predict_proba([[3.2, 2.5]])[0][1]
+
+print('Triangle:', proba_triangle)
+print('Square:', proba_square)
+print('Circle:', proba_circle)
+```
+
+    Triangle: 0.1457476378273223
+    Square: 0.28507914730638007
+    Circle: 0.49761172107185986
+
+
+triangle | square   | circle
+---      | ---      | ---
+0.145748 | 0.285079 | 0.497612
+
+Since the positive class probability of `circle` is the greatest of the three, our multiclass classifier predicts that the observation is a `circle`.
+
+## Case Study: Iris dataset
+
+The [Iris dataset](https://archive.ics.uci.edu/ml/datasets/iris) is a famous dataset that is often used in data science to explore machine learning concepts. There are three classes, each representing a type of Iris plant:
+- Iris-setosa (represented in the dataset by `0`)
+- Iris-versicolour (`1`)
+- Iris-virginica (`2`)
+
+There are four features available in the dataset:
+- Sepal length (cm)
+- Sepal width (cm)
+- Petal length (cm)
+- Petal width (cm)
+
+![](petal_sepal.png)
+
+We will create a multiclass classifier that predicts the type of Iris plant based on the four features above. First, we read in the data:
+
+
+```python
+iris = pd.read_csv('https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data',
+                  header=None, names=['sepal_length', 'sepal_width', 'petal_length', 'petal_width', 'species'])
+
+iris.head()
```

Comment:
With the additional pandas config above, you can just display `iris` for a nicer output (it shows the total number of rows too).